collated/main/fustilio.md

```diff
@@ -0,0 +1,581 @@
+     * @throws DuplicateParcelException if editedParcel already exists unless the parcelToEdit is the same entity.
+     * @throws ParcelNotFoundException if parcelToEdit cannot be found in the list
+     */
+    void editParcelCommand(ReadOnlyParcel parcelToEdit, ReadOnlyParcel editedParcel)
+            throws DuplicateParcelException, ParcelNotFoundException;
+
+    /**
+     * Method to retrieve flag that represents whether the current tab selected is all parcels.
+     */
+    boolean getActiveIsAllBool();
+
+    /**
+     * Method to forcefully raise the event to switch tabs to all parcels.
+     */
+    void uiJumpToTabAll();
+
+    /**
+     * Method to forcefully rasie the event to switch tabs to completed parcels.
+     */
+    void uiJumpToTabCompleted();
+```
+###### \java\seedu\address\model\ModelListener.java
+``` java
+package seedu.address.model;
+
+import com.google.common.eventbus.Subscribe;
+
+import seedu.address.commons.core.EventsCenter;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.events.ui.JumpToTabRequestEvent;
+
+/**
+ * SelectionListener listens for events that select a parcel card.
+ */
+public class ModelListener {
+
+    private Model model = null;
+
+    /**
+     * Initializes a SelectionLister with the given model.
+     */
+    public ModelListener(Model model) {
+        this.model = model;
+        registerAsAnEventHandler(this);
+    }
+
+    /**
+     * Registers the object as an event handler at the {@link EventsCenter}
+     * @param handler usually {@code this}
+     */
+    protected void registerAsAnEventHandler(Object handler) {
+        EventsCenter.getInstance().registerHandler(handler);
+    }
+
+    /**
+     * Triggers when there is a JumpToTabRequestEvent and sets the tabIndex in the model
+     * to keep track of which tab the model is "on".
+     */
+    @Subscribe
+    private void handleJumpToTabEvent(JumpToTabRequestEvent event) {
+        model.setTabIndex(Index.fromZeroBased(event.targetIndex));
+    }
+}
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /** Deletes the tag from every parcel in the address book */
+    public void deleteTag(Tag target) throws TagNotFoundException, TagInternalErrorException {
+
+        int tagsFound = 0;
+        Iterator it = addressBook.getParcelList().iterator();
+        while (it.hasNext()) {
+            Parcel oldParcel = (Parcel) it.next();
+            Parcel newParcel = new Parcel(oldParcel);
+            Set<Tag> newTags = new HashSet<>(newParcel.getTags());
+            if (newTags.contains(target)) {
+                newTags.remove(target);
+                tagsFound++;
+            }
+
+            newParcel.setTags(newTags);
+
+            try {
+                addressBook.updateParcel(oldParcel, newParcel);
+            } catch (DuplicateParcelException | ParcelNotFoundException dpe) {
+                throw new TagInternalErrorException();
+            }
```

Comment:
Perhaps line 263 - 278 can be abstracted into a method since the `deleteTag` method is a little long.